conformance/base/manifests.yaml

```diff
@@ -223,6 +223,42 @@ spec:
 ---
 apiVersion: v1
 kind: Service
+metadata:
```

Comment:
do we need a new service and deployment for this case ?
the delay is being instrumented using the request query param

Reply:
It's a different echo service (different image). It's the echo service used for mesh testing, which supports delay. The other services used for gateway testing are using a different echo service, which do not support the query param. It's also nice to have a specific named service for the delaying backend, for readability, IMO.

Reply:
maybe you just want to bump the echo sever used in the project which is actually the istio app
https://github.com/kubernetes-sigs/gateway-api/blob/main/docker/Dockerfile.echo

Reply:
also most implementation are running conformance tests on CI with minimal resources such as GH Runners, so prefer less resources to be instantiated

Reply:
Maybe I'm missing something, but I thought that there were 2 different echo services with different implementations.

1. `gcr.io/k8s-staging-ingressconformance/echoserver:v...` used in `base/manfiest.yaml`
2. `gcr.io/k8s-staging-gateway-api/echo-server:v...` used in `mesh/manifest.yaml`

My understanding is that the second one comes from Istio and supports delay, but the first one does not.

Are you suggesting that we should switch to use the Istio one for everything in both manifest.yaml files? If so, then we don't need a separate service for the timeout test, we could just use backend-v1, for example, with the delay param?

I think that if I'm understanding you, I agree that would be better, but I'd rather leave that for a separate cleanup PR, instead of changing all of that as part of this one. WDYT?

Reply:
Sorry I missed this! It probably is simpler to standardize around a single base image, so I'm fine with this change. I think it's definitely preferable over adding new Pods (and by extension increasing the resources required to run conformance tests). Maybe for this PR just swap one existing deployment over to the newer image and then we can have a follow up to get the rest.

Reply:
Turns out that the two echo services are very different. The echo service used by the gateway conformance framework (in `base/manifest.yaml`) sends back `application/json` content containing all of the http request parameters. The Istio one just sends back whatever was in the request body. The gateway-api conformance framework is very much dependent on the `application/json` response, it's even a little difficult to write a test that uses a different target service. The test case needs to set `ExpectedRequest: &http.ExpectedRequest{}` to explicitly disable checking the response for the `application/json` that's expected from the echo service:
```
	testCases := []http.ExpectedResponse{
			{
				Request:         http.Request{Path: "/backend-timeout"},
				ExpectedRequest: &http.ExpectedRequest{},
				Response:        http.Response{StatusCode: 200},
			}, {
```

Anyway, I think that if we want to simplify things, we could add delay support to the gateway echo service, so that we can use it for all the gateway tests (i.e., `base/manifest.yaml`) but I think we'll probably still need to use the other (Istio) echo service for the mesh testing (`mesh/manifest.yaml`).

Reply:
Thanks for looking into this! Agree that it's likely out of scope to switch everything over to a different base image here. Not sure if you've got time, but it may be relatively straightforward to add support for a `delay` query param to the main image we're using, probably somewhere around here: https://github.com/kubernetes-sigs/ingress-controller-conformance/blob/master/images/echoserver/echoserver.go#L142. 

Longer term, I think it would be good to bring this into the Gateway API repo so we don't have to go to another mostly frozen repo to make small adjustments like this. If you want to give that a shot, I've created https://github.com/kubernetes-sigs/gateway-api/issues/2445 to track it.

Reply:
> it may be relatively straightforward to add support for a delay query param to the main image we're using, probably somewhere around here: https://github.com/kubernetes-sigs/ingress-controller-conformance/blob/master/images/echoserver/echoserver.go#L142.

Yes, I agree that's where to add it. Just need to add basically the same code that Istio has here: https://github.com/istio/istio/blob/da3bda9fd4bcbfc1873e0baab6eaddb4ad50aebe/pkg/test/echo/server/endpoint/http.go#L254

@SRodi or I will make sure to do this and #2445 relatively soon.

As far as this PR is concerned, I would like to get it merged now, if possible, since it works and I'd like to have it running in the Istio build integration testing. I'll definitely take the TODO to send a followup PR to clean it up to use the main image as soon as it's available with the delay function. WDYT?